ddtrace/tracer/remote_config.go

```diff
@@ -236,6 +286,44 @@ func (t *tracer) onRemoteConfigUpdate(u remoteconfig.ProductUpdate) map[string]s
 	return statuses
 }
 
+// Handle enabling or disabling of Dynamic Instrumentation / Live Debugger.
+//
+// Returns a telemetry configuration if the value changed.
+func (t *tracer) handleDynamicInstrumentationEnabledRC(val *bool) *telemetry.Configuration {
+	// Do not overwrite a "false" value coming from the
+	// DD_DYNAMIC_INSTRUMENTATION_ENABLED env var, or from other local sources.
+	finalSources := []telemetry.Origin{telemetry.OriginCode, telemetry.OriginEnvVar, telemetry.OriginDDConfig}
+	if !t.config.dynamicInstrumentationEnabled.get() &&
+		slices.Contains(finalSources, t.config.dynamicInstrumentationEnabled.cfgOrigin) {
+		return nil
+	}
+
+	if !t.config.dynamicInstrumentationEnabled.handleRC(val) {
+		return nil
+	}
+
+	// The value changed; subscribe or unsubscribe from the Live Debugging RC
+	// product.
+	if t.config.dynamicInstrumentationEnabled.get() {
```

Comment:
Just confirming that line 301 has the potential to change the value of `t.config.dynamicInstrumentationEnabled.get()`? Otherwise, this condition is always true, as we already returned for `!t.config.dynamicInstrumentationEnabled.get()` on line 298.